### PR TITLE
Feature/reward delivery confirmation email

### PR DESCRIPTION
**Pull request by K4sku**
Sprint 5 / task 3

Big one, comes with hosting and cosmetics changes:

**Main features:**

- add configuration for ActionMailer to send emails via SendGrid API
- add SendGrid API key to credentials
- set RAILS_MASTER_KEY in Heroku 
- add OrderMailer that sends order confirmation emails
- add sending email to employee when admin marks order as delivered via ActiveJob
- add unit test of this mailing functionality
- add [premailer-rails](https://github.com/fphilipe/premailer-rails) gem; it inlines CSS to emails views and generates text version from html so they are always in sync
- add [letter_opener](https://github.com/ryanb/letter_opener) gem to development and text environments
- updates Gemfile.lock
- updates seeds; change users email domain to @cklos.com to test mailing functionality on prod

Fixes:

- remove option to recover password for admins
- fix incorrectly testing orders placement date in tests: 
    spec/system/admins/employee/show_spec.rb, 
    spec/system/admins/orders/deliver_order_spec.rb, 
    spec/system/orders/index_spec.rb

Cosmetics:

- update readme.md
- style devise's password reset template for employee
- fix typo in logo (was "ERC" instead of "ERP")

Notes:

- add custom domain www.erp.cklos.com - unfortunately using HTTPS on custom domains requires paid plan on Heroku
- emails sent to seeded users are forwarded via cloudflare to designated email
- add Heroku staging pipeline to app; Pull Requests that pass CI actions are deployed automatically, link can be found below. 

order-delivery
NOTE: link change in GIF is result of using Heroku staging pipeline and will not occur once app hit production

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,6 +52,7 @@ group :development do
   gem 'listen', '~> 3.3'
   gem 'rack-mini-profiler', '~> 2.0'
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
```
Comment (piowit) [inline Gemfile]: we can remove this comment as right now it is not near Spring, and also we don't need it :)

File: app/models/admin.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class Admin < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  # :registerable
+  # :registerable, :recoverable,
   devise :database_authenticatable,
-         :recoverable, :rememberable, :validatable
+         :rememberable, :validatable
```
Comment (piowit) [inline app/models/admin.rb:6]: I think that this is not needed and it is safer to recover the admin account by changing the password using a console, then giving phishers the ability to recover via email :)

File: spec/mailers/previews/order_preview.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Preview all emails at http://localhost:3000/rails/mailers/order
+class OrderPreview < ActionMailer::Preview
+  def order_delivered_email
+    order = Order.first
+    OrderMailer.with(order: order).order_delivered_email
```
Comment (piowit) [inline spec/mailers/previews/order_preview.rb:5]: What happens if there will be no Orders in DB?

**Review by piowit — COMMENTED** [review]
👍🏻 Looks ok
I left few comments below

File: spec/mailers/previews/order_preview.rb (hunk 3 above)
Comment (K4sku) [reply spec/mailers/previews/order_preview.rb:5]: It does not load order object and crashes with undefined method for nil. I have changed it to use factory to provide object

File: Gemfile (hunk 1 above)
Comment (K4sku) [reply Gemfile]: I have removed unnecessary comments and order gems alphabetically

File: spec/mailers/previews/order_preview.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Preview all emails at http://localhost:3000/rails/mailers/order
+# loads FactoryBot to use build strategy
+class OrderPreview < ActionMailer::Preview
+  include FactoryBot::Syntax::Methods
+
+  def order_delivered_email
+    order = build(:order)
```
Comment (ciejj) [inline spec/mailers/previews/order_preview.rb]: how bout:
order = FactoryBot.build(:order)
you do not have to include FactoryBot then

**Review by ciejj — APPROVED** [review]
Nice one.

Just a small request - please add refactors to separate PR - you can ask any of the participants, or me for checking that additional PR, and it will make review process faster ;)

File: spec/mailers/previews/order_preview.rb (hunk 4 above)
Comment (K4sku) [reply spec/mailers/previews/order_preview.rb]: That's good idea. I have not though of that.

**Pull request merged**